src/StackExchange.Metrics/IMetricsCollector.cs

```diff
@@ -0,0 +1,96 @@
+        /// <param name="description">The metadata description of the metric.</param>
+        /// <param name="metricFactory">A delegate which will be called to instantiate the metric.</param>
+        T GetMetric<T>(string name, string unit, string description, Func<T> metricFactory) where T : MetricBase;
+
+        /// <summary>
+        /// Creates a metric (time series) and adds it to the collector. If a metric by the same name and tag values already exists, then that metric is
+        /// returned. The <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will be prepended to the metric name.
+        /// </summary>
+        /// <param name="name">The metric name. The global prefix <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will be prepended.</param>
+        /// <param name="unit">The units of the metric (e.g. "milliseconds").</param>
+        /// <param name="description">The metadata description of the metric.</param>
+        /// <param name="metric">A pre-instantiated metric, or null if the metric type has a default constructor.</param>
+        T GetMetric<T>(string name, string unit, string description, T metric = null) where T : MetricBase;
+
+        /// <summary>
+        /// Creates a metric (time series) and adds it to the collector. If a metric by the same name and tag values already exists, then that metric is
+        /// returned. The <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will NOT be prepended to the metric name.
+        /// </summary>
+        /// <param name="name">The metric name. The global prefix <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will be prepended.</param>
+        /// <param name="unit">The units of the metric (e.g. "milliseconds").</param>
+        /// <param name="description">The metadata description of the metric.</param>
+        /// <param name="metricFactory">A delegate which will be called to instantiate the metric.</param>
+        T GetMetricWithoutPrefix<T>(string name, string unit, string description, Func<T> metricFactory) where T : MetricBase;
+
+        /// <summary>
+        /// Creates a metric (time series) and adds it to the collector. If a metric by the same name and tag values already exists, then that metric is
+        /// returned. The <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will NOT be prepended to the metric name.
+        /// </summary>
+        /// <param name="name">The metric name. The global prefix <see cref="MetricsCollectorOptions.MetricsNamePrefix"/> will be prepended.</param>
+        /// <param name="unit">The units of the metric (e.g. "milliseconds").</param>
+        /// <param name="description">The metadata description of the metric.</param>
+        /// <param name="metric">A pre-instantiated metric, or null if the metric type has a default constructor.</param>
+        T GetMetricWithoutPrefix<T>(string name, string unit, string description, T metric = null) where T : MetricBase;
```

Comment:
Since this is 2.x, would it be a good time to break this method? Proposal: we reduce the methods by half by include a default for `bool includePrefix = true` instead of having a duple for each method.

Thoughts?